### PR TITLE
Pre release improvements

### PR DESCRIPTION
**Pull request by inverno**

### CODE REVIEW

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.InputStreamReader;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.sonar.api.batch.BatchSide;
+import org.sonar.api.batch.ScannerSide;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+
+@BatchSide
```
Comment (SonarTech) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: MINOR Remove this use of "BatchSide"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@ScannerSide
+@SonarLintSide
+public class ExternalProcessErrorConsumer {
+
+  // Available to subclasses
+  protected static final Logger LOG = Loggers.get(ExternalProcessErrorConsumer.class);
```
Comment (SonarTech) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: MAJOR Make the "LOG" logger private static final. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  public final void consumeStdError(Process process) {
+    final ExecutorService errorConsumer = Executors.newSingleThreadExecutor();
+    errorConsumer.submit(() -> {
+      try (BufferedReader errorReader = new BufferedReader(new InputStreamReader(process.getErrorStream()))) {
```
Comment (SonarTech) [inline sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: MAJOR Remove this use of constructor "InputStreamReader(InputStream)" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonarts-core/yarn.lock [hunk 4]
```diff
@@ -150,13 +150,7 @@ ansi-styles@^2.2.1:
   version "2.2.1"
   resolved "https://registry.yarnpkg.com/ansi-styles/-/ansi-styles-2.2.1.tgz#b432dd3358b634cf75e1e4664368240533c1ddbe"
 
-ansi-styles@^3.0.0:
```
Comment (vilchik-elena) [inline sonarts-core/yarn.lock]: @inverno why changes to this file?

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java (hunk 1 above)
Comment (vilchik-elena) [reply sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: @inverno why all this annotations?

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java (hunk 3 above)
Comment (vilchik-elena) [reply sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: @inverno whyt about this issue?

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java (hunk 1 above)
Comment (inverno) [reply sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: I need to inject this object in the sensor at the plug-in level

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java (hunk 3 above)
Comment (inverno) [reply sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: Likely correct, I think the charset is missing

File: sonarts-core/yarn.lock (hunk 4 above)
Comment (inverno) [reply sonarts-core/yarn.lock]: I have no clue, I just ran build.sh but back then I had other fish to fry and I wanted to see if QA had similar build problems as I was experiencing

File: sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java (hunk 1 above)
Comment (inverno) [reply sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalProcessErrorConsumer.java]: If you mean specifically 'why BatchSide' it's for backwards compatibility afaik

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**